browser/brave_vpn/win/brave_vpn_wireguard_service/service/brave_wireguard_manager.cc

```diff
@@ -6,19 +6,69 @@
 #include "brave/browser/brave_vpn/win/brave_vpn_wireguard_service/service/brave_wireguard_manager.h"
 
 #include <string>
+
+#include "base/base64.h"
 #include "base/logging.h"
 #include "base/strings/utf_string_conversions.h"
 #include "brave/browser/brave_vpn/win/brave_vpn_wireguard_service/service/wireguard_tunnel_service.h"
+#include "brave/components/brave_vpn/common/wireguard/wireguard_utils.h"
 
 namespace brave_vpn {
 
-HRESULT BraveWireguardManager::EnableVpn(const wchar_t* config,
+HRESULT BraveWireguardManager::EnableVpn(BSTR public_key,
+                                         BSTR private_key,
+                                         BSTR address,
+                                         BSTR endpoint,
                                          DWORD* last_error) {
-  if (!config || !last_error) {
-    VLOG(1) << "Invalid parameters";
+  if (!last_error) {
+    VLOG(1) << "last_error must be set";
+    return E_FAIL;
+  }
+
+  // if all params are empty, reconnect using last known good config.
+  // browser/brave_vpn/win/brave_vpn_wireguard_service/service/wireguard_tunnel_service.cc
+  bool reconnect_using_last_config = public_key && wcslen(public_key) == 0 &&
+                                     private_key && wcslen(private_key) == 0 &&
+                                     address && wcslen(address) == 0 &&
+                                     endpoint && wcslen(endpoint) == 0;
+  if (reconnect_using_last_config) {
+    *last_error = !brave_vpn::wireguard::LaunchWireguardService(L"");
+    return S_OK;
+  }
+
+  std::string public_key_str;
+  if (!brave_vpn::wireguard::ValidateKey(public_key, &public_key_str,
+                                         "public_key")) {
+    return E_FAIL;
+  }
+
+  std::string private_key_str;
+  if (!brave_vpn::wireguard::ValidateKey(private_key, &private_key_str,
+                                         "private_key")) {
     return E_FAIL;
   }
-  *last_error = !brave_vpn::wireguard::LaunchWireguardService(config);
+
+  std::string address_str;
```

Comment:
Can you remember me what the `address_str` is used for?
Isn't the `endpoint` the only required parameter to start a connection?

Reply:
This is the address the VPN host assigned when getting the credentials